### DISCUSSION TITLE
Awaiting current user with Nuxt3

### DISCUSSION THREAD

**Discussion by Cry0nicS** [vuejs/vuefire#1557] [Questions and Help]
### Reproduction

I provided a link below to my repository. Not sure how to reproduce with an actual Firebase database...

### Steps to reproduce the bug

Using Nuxt3 with SSR and vuefire.

This is what I use on a page - I first await the user, then I load his  Collection

```ts
useFirebaseAuth();
const user = (await getCurrentUser()) as User;
const db = useFirestore();

const userCollectionsQuery = query(
    collection(db, user.uid as string),
    orderBy("sessionStartedAt", "desc")
);

const {data: userCollections} = useCollection<UserCollection>(userCollectionsQuery, {
    once: true,
    ssrKey: user.uid
});
```

### Expected behavior

I would expect for the page to first wait until I have a user and then proceed.

### Actual behavior

Seems like there's either some sort of race condition or some other sort of issue, where the user is not await.
The middleware (as per the document) deals with `await getCurrentUser()` perfectly, but somehow on a Nuxt3 page this doesn't work.
It basically throws a 500 `Cannot read properties of null (reading 'uid')` (on user.uid) or something similar.

With a bunch of refreshes, sometimes it works. Most of the time, it doesn't.

### Additional information

I don't have a reproduction, since I'm not sure how to provide a connection to Firebase. But here's a permalink to my project where I use exactly that. [Link](https://github.com/Cry0nicS/cancer-classifier-frontend/blob/891e3a99c9ed45ed4c73fac804e394e4d31f2167/pages/services/history.vue#L29 )

The reason I want to use `await getCurrentUser` instead of `useCurrentUser` is to avoid all the optional chaining and to... well, only display the page when the user is already loaded.

**Comment by dword-design** [comment]
You can set a [session cookie](https://vuefire.vuejs.org/nuxt/auth.html#Session-Cookie), then the user will already be there SSR.
